File: AngularJS201/AngularJS201/Controllers/PermissionController.cs
using AngularJS201.app.Model;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Data;
using System.IO;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Xml;
using System.Xml.Linq;

namespace AngularJS201.Controllers
{
    public class PermissionController : Controller
    {
        // GET: Page
        public ActionResult Index()
        {
            return View();
        }

        [HttpPost]
        public void WritePermissionData(List<Roles> roles)
        {
            WritePermissionDataXml(roles);             
        }
                
        [HttpGet]
        public string ReadPermissionData()
        {
            return ReadPermissionDataXml();
        }
        
        private void WritePermissionDataXml(List<Roles> roles)
        {
            string filepath = Server.MapPath("~/App_Data/role.xml");
            XDocument doc = XDocument.Load(Server.MapPath("~/App_Data/role.xml"));

            foreach (var role in roles)
            {
                XElement roleCollection = doc.Elements("roles").Elements("role")
                                              .Where(x => x.Element("RoleName").Value == role.RoleName)
                                              .FirstOrDefault();
                 
                roleCollection.Element("Permission").SetValue(role.Permission);
            } 
            doc.Save(filepath);         
           
        }

        private string ReadPermissionDataXml()
        {
            string result = string.Empty;
            XmlDocument doc = new XmlDocument();
            try
            {
                doc.Load(Server.MapPath("~/App_Data/role.xml"));
            }
            catch
            {
                ;
            }

            XmlNodeList elemList = doc.GetElementsByTagName("role");
            var permissions = new List<Roles>();
            for (int i = 0; i < elemList.Count; i++)
            {
                XmlNode xmlnodeRoleName = elemList[i].FirstChild;
                XmlNode xmlnodeDescription = elemList[i].ChildNodes[1];
                XmlNode xmlnodePermission = elemList[i].LastChild;
                permissions.Add(
                 new Roles()
                 {
                     RoleName = xmlnodeRoleName.InnerText,
                     Description = xmlnodeDescription.InnerText,
                     Permission = xmlnodePermission.InnerText
                 });
            }

            if (permissions != null)
                result = JsonConvert.SerializeObject(permissions).ToString();
            return result;

        }

    }
}

File: AngularJS201/AngularJS201/app/Controllers/PermissionController.js
(function () {
    'use strict';

    angular
        .module('app')
        .controller('PermissionController', ['$scope', 'permissionDataService', function ($scope, permissionDataService) {
            $scope.title = "Permission";    

            $scope.test = function () {
                
                angular.forEach($scope.rolePermissions, function (value, key) {
                    let temp = "";
                    if (value.R ) {
                        temp += "R";
                    }
                    if (value.W) {
                        temp += "W";
                    }
                    if (value.U) {
                        temp += "U";
                    }
                    if (value.D) {
                        temp += "D";
                    }
                    console.log(temp);
                    $scope.permissions[key].Permission = temp 
                });  
            };


            $scope.saveData = function () {
                
                angular.forEach($scope.rolePermissions, function (value, key) {
                    let R = (value.R == true || value.R != -1) ? "R" : "";
                    let W = (value.W == true || value.W != -1) ? "W" : "";
                    let U = (value.U == true || value.U != -1) ? "U" : "";
                    let D = (value.D == true || value.D != -1) ? "D" : "";
                    let tempPermissions = R + W + U + D;
                    $scope.rolePermissions[key].Permission = tempPermissions; 
                });
                let permissions = permissionDataService.savePermissionData($scope.rolePermissions,"NA").then(function (response) {
                    return;
                });
            };          



            $scope.readData = function () {   
                let permissions = permissionDataService.readPermissionData().then(function (response) {
                   
                    $scope.permissions = response;
                    $scope.rolePermissions=[];
                    angular.forEach($scope.permissions, function (value, key) {
                        let temp = {
                            RoleName: value.RoleName,
                            Permission: value.Permission,
                            R: value.Permission.indexOf('R'),
                            W: value.Permission.indexOf('W'),
                            U: value.Permission.indexOf('U'),
                            D: value.Permission.indexOf('D')
                        };  
                        $scope.rolePermissions.push(temp);
                        
                    });
                                                         
                });               
            };
            
           $scope.readData();

        }]);
})();